src/runtime-rs/crates/agent/src/types.rs

```diff
@@ -390,6 +390,32 @@ pub struct KernelModule {
     pub parameters: Vec<String>,
 }
 
+impl KernelModule {
+    // input string: "ModuleName Param1 Param2 ..."
+    pub fn new(str: String) -> Self {
+        let split: Vec<&str> = str.split(' ').collect();
```

Comment:
Although the current module handling doesn't support it, it's worth pointing out that this won't work for all modules. From https://github.com/torvalds/linux/blob/master/Documentation/admin-guide/kernel-parameters.rst:

> Double-quotes can be used to protect spaces in values, e.g.:
>
> `param="spaces in here"`

Reply:
Do we need to add this part of the implementation to the existing module handling as well? 🤔

Reply:
Yes, this is indeed a corner case.  I can think of a scenario where this would be required. For example, If we implement a module with a message to be printed, and this message is passed as parameter, then we should consider the "space". 🙃